### ISSUE TITLE
``expression has no type`` error

### ISSUE THREAD

**Issue by vegansk** [nim-lang/Nim#5421]
```nim
type A[T: int|string] = object
    x: T

proc initA[T](x: T): A[T] =
  result.x = x

let a = initA(1)
```
compilation fails with error:
```
terr.nim(7, 14) Error: expression 'initA(1)' has no type (or is ambiguous)
```

**Comment by andreaferretti** [comment]
What version are you using? Seems fixed on both v0.17 and on devel

**Comment by vegansk** [comment]
Yes, it's fixed already.
